android/src/main/java/com/rncamerakit/CKCamera.kt

```diff
@@ -307,9 +308,42 @@ class CKCamera(context: ThemedReactContext) : FrameLayout(context), LifecycleObs
         val useCases = mutableListOf(preview, imageCapture)
 
         if (scanBarcode) {
-            val analyzer = QRCodeAnalyzer { barcodes ->
-                if (barcodes.isNotEmpty()) {
-                    onBarcodeRead(barcodes)
+            val analyzer = QRCodeAnalyzer { results ->
+                if (results.isNotEmpty()) {
+                    val filteredResults = if (barcodeFrame !== null) {
```

Comment:
We rarely, if ever, use triple equality in Kotlin. Same below with `===`, prefer `==` instead

You can also create a shadow copy of `barcodeFrame`. This is not very sexy, but `barcodeFrame` is now a local `val`, and the compiler will allow using it in the `if` as a non optional.

Lastly, reverting the condition makes the code easier to read as we avoid having a one line else 20 lines down

```suggestion
                    val barcodeFrame = barcodeFrame
                    val filteredResults = if (barcodeFrame == null) {
                        results
                    } else {
```